eng/Subsets.props

```
@@ -50,10 +50,14 @@
     <_NativeAotSupportedArch Condition="'$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'loongarch64' or '$(TargetArchitecture)' == 'riscv64' or ('$(TargetOS)' == 'windows' and '$(TargetArchitecture)' == 'x86')">true</_NativeAotSupportedArch>
     <NativeAotSupported Condition="'$(_NativeAotSupportedOS)' == 'true' and '$(_NativeAotSupportedArch)' == 'true'">true</NativeAotSupported>
 
-    <!-- Determine if AOT tools can run on the specified target -->
-    <_AotToolsSupportedOS Condition="'$(TargetsMobile)' != 'true' and '$(TargetsLinuxBionic)' != 'true' and '$(StageOneBuild)' != 'true'">true</_AotToolsSupportedOS>
-    <_AotToolsSupportedArch Condition="'$(TargetArchitecture)' != 'armel'">true</_AotToolsSupportedArch>
-    <AotToolsSupported Condition="'$(_AotToolsSupportedOS)' == 'true' and '$(_AotToolsSupportedArch)' == 'true'">true</AotToolsSupported>
+    <!-- Determine if we support running the .NET SDK on the target platform -->
+    <_SdkToolsSupportedOS Condition="'$(TargetsMobile)' != 'true' and '$(TargetsLinuxBionic)' != 'true'">true</_SdkToolsSupportedOS>
+    <_SdkToolsSupportedArch Condition="'$(TargetArchitecture)' != 'armel'">true</_SdkToolsSupportedArch>
+    <SdkToolsSupported Condition="'$(_SdkToolsSupportedOS)' == 'true' and '$(_SdkToolsSupportedArch)' == 'true'">true</SdkToolsSupported>
+
+    <!-- Delete `_IsCommunityCrossArchitecture` once https://github.com/dotnet/runtime/issues/104661 is fixed. -->
+    <_IsCommunityCrossArchitecture Condition="'$(CrossBuild)' == 'true' and '$(TargetArchitecture)' == 'loongarch64'">true</_IsCommunityCrossArchitecture>
+    <UseNativeAotForComponents Condition="'$(NativeAotSupported)' == 'true' and '$(TargetOS)' == '$(HostOS)' and '$(TargetsLinuxBionic)' != 'true' and '$(_IsCommunityCrossArchitecture)' != 'true'">true</UseNativeAotForComponents>
```

```suggestion
    <UseNativeAotForComponents Condition="'$(NativeAotSupported)' == 'true' and '$(TargetOS)' == '$(HostOS)' and '$(TargetsLinuxBionic)' != 'true'">true</UseNativeAotForComponents>
```

@LuckyXu-HF has confirmed that LA64's PublishAot is ready for prime time https://github.com/dotnet/runtime/pull/114618#discussion_r2044330666.

---

Ah need to update target.

Option 1: after this line https://github.com/dotnet/runtime/blob/0b066ede7a9376fee58f926ebbaf0aea762279ab/eng/toolAot.targets#L51 add:

```xml
<!-- CAS-128 requires ISA 1.1 (la664). Force -march=la664 until cross toolchains defaults to 1.1. -->
<LinkerArg Include="-march=la664" Condition="$(TargetArchitecture) == 'loongarch64' and '$(CrossBuild)' == 'true'" />
```
This is based on https://github.com/dotnet/runtime/pull/108364#issuecomment-2381132048.

Option 2:  add loongarch64 in this condition: https://github.com/dotnet/runtime/blob/522cf16428cc23c71cf3ceb7a644c305003b8870/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets#L225

I like option 2, because on ISA1.1, libatomic makes it pass-thru (to hardware instruction) and supports ISA1.0. The only overhead is +1 library, which is minuscule (besides it's for time-being). However, since it's working for @LuckyXu-HF and @shushanhf on the actual device (no cross-build), perhaps option 1 would be their preference?

---

Ah need to update target.

Option 1: after this line https://github.com/dotnet/runtime/blob/0b066ede7a9376fee58f926ebbaf0aea762279ab/eng/toolAot.targets#L51 add:

```xml
<!-- CAS-128 requires ISA 1.1 (la664). Force -march=la664 until cross toolchains defaults to 1.1. -->
<LinkerArg Include="-march=la664" Condition="$(TargetArchitecture) == 'loongarch64' and '$(CrossBuild)' == 'true'" />
```
This is based on https://github.com/dotnet/runtime/pull/108364#issuecomment-2381132048.

Option 2:  add loongarch64 in this condition: https://github.com/dotnet/runtime/blob/522cf16428cc23c71cf3ceb7a644c305003b8870/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets#L225

I like option 2, because on ISA1.1, libatomic makes it pass-thru (to hardware instruction) and provides software fallback for ISA1.0. The only overhead is +1 library linkage, which is minuscule (besides it's for time-being). However, since it's working for @LuckyXu-HF and @shushanhf on the actual device (no cross-build), perhaps option 1 would be their preference?

---

Ops march is a compiler option, so it will be after: https://github.com/dotnet/runtime/blob/522cf16428cc23c71cf3ceb7a644c305003b8870/eng/native/configurecompiler.cmake#L779

```cmake
if(CLR_CMAKE_HOST_ARCH_LOONGARCH64 AND CMAKE_CROSSCOMPILING)
   # CAS-128 requires ISA 1.1 (la664). Force -march=la664 until toolchains default to 1.1 (or above).
   add_compile_options(-march=la664)
endif()
```